pkgs/by-name/xo/xonsh/unwrapped.nix

```diff
@@ -1,106 +1,120 @@
+      # fails on sandbox
+      "tests/completers/test_command_completers.py"
+      "tests/test_ptk_highlight.py"
+      "tests/test_ptk_shell.py"
+      # fails on non-interactive shells
+      "tests/prompt/test_gitstatus.py"
+      "tests/completers/test_bash_completer.py"
+    ];
 
-  disabledTests = [
-    # fails on sandbox
-    "test_colorize_file"
-    "test_loading_correctly"
-    "test_no_command_path_completion"
-    "test_bsd_man_page_completions"
-    "test_xonsh_activator"
-    # fails on non-interactive shells
-    "test_capture_always"
-    "test_casting"
-    "test_command_pipeline_capture"
-    "test_dirty_working_directory"
-    "test_man_completion"
-    "test_vc_get_branch"
-    "test_bash_and_is_alias_is_only_functional_alias"
-  ];
+    nativeCheckInputs = [
+      git
+      glibcLocales
+    ] ++ (with python3.pkgs; [
+      pip
+      pyte
+      pytest-mock
+      pytest-subprocess
+      pytestCheckHook
+    ]);
 
-  disabledTestPaths = [
-    # fails on sandbox
-    "tests/completers/test_command_completers.py"
-    "tests/test_ptk_highlight.py"
-    "tests/test_ptk_shell.py"
-    # fails on non-interactive shells
-    "tests/prompt/test_gitstatus.py"
-    "tests/completers/test_bash_completer.py"
-  ];
+    postPatch = ''
+      sed -ie "s|/bin/ls|${coreutils}/bin/ls|" tests/test_execer.py
+      sed -ie "s|SHELL=xonsh|SHELL=$out/bin/xonsh|" tests/test_integrations.py
 
-  nativeCheckInputs = [
-    git
-    glibcLocales
-  ] ++ (with python3.pkgs; [
-    pip
-    pyte
-    pytest-mock
-    pytest-subprocess
-    pytestCheckHook
-  ]);
+      sed -ie 's|/usr/bin/env|${coreutils}/bin/env|' tests/test_integrations.py
+      sed -ie 's|/usr/bin/env|${coreutils}/bin/env|' scripts/xon.sh
 
-  preCheck = ''
-    export HOME=$TMPDIR
-  '';
+      find scripts -name 'xonsh*' -exec sed -i -e "s|env -S|env|" {} \;
+      find -name "*.xsh" | xargs sed -ie 's|/usr/bin/env|${coreutils}/bin/env|'
+      patchShebangs .
+    '';
 
-  passthru = {
-    shellPath = "/bin/xonsh";
-    python = python3; # To the wrapper
-    updateScript = gitUpdater { };
-  };
+    preCheck = ''
+      export HOME=$TMPDIR
+    '';
 
-  meta =  {
-    homepage = "https://xon.sh/";
-    description = "A Python-ish, BASHwards-compatible shell";
-    changelog = "https://github.com/xonsh/xonsh/raw/${version}/CHANGELOG.rst";
-    license = with lib.licenses; [ bsd3 ];
-    maintainers = with lib.maintainers; [ AndersonTorres ];
+    passthru = {
+      shellPath = "/bin/xonsh";
+      python = python3; # To the wrapper
+      updateScript = gitUpdater { };
+    };
+
+    meta = {
+      homepage = "https://xon.sh/";
+      description = "A Python-ish, BASHwards-compatible shell";
+      changelog = "https://github.com/xonsh/xonsh/raw/${argset.version}/CHANGELOG.rst";
+      license = with lib.licenses; [ bsd3 ];
+      maintainers = with lib.maintainers; [ AndersonTorres ];
+    };
   };
-}
+in
+python3.pkgs.buildPythonApplication argset
```

Comment:
Wait, why move everything to `argset`?

Reply:
I am using an explicitly self-referent set; in other words, a rec without rec.

Reply:
Ah, get it. I don't expect `overrideAttrs` to work, though.